src/pages/ContributionsPage/ContributionForm.tsx

```diff
@@ -0,0 +1,209 @@
+        circle_id: circleId,
+        description: value,
+      });
+    };
+  }, [currentContribution?.contribution.id]);
+
+  const updateSaveStateForContribution = (
+    id: number | undefined,
+    saveState: SaveState
+  ) => {
+    if (id == undefined) {
+      return;
+    }
+    setSaveState(prevState => {
+      const newState = { ...prevState };
+      newState[id] = saveState;
+      return newState;
+    });
+  };
+
+  return (
+    <>
+      <ContentHeader>
+        <Flex column css={{ width: '100%' }}>
+          <Flex column alignItems="end" css={{ gap: '$sm' }}>
+            {showMarkdown ? (
+              <Box
+                tabIndex={0}
+                css={{ borderRadius: '$3', width: '100%' }}
+                onClick={() => setShowMarkDown(false)}
+                onKeyDown={e => {
+                  e.stopPropagation();
+                  if (e.key === 'Enter' || e.key === ' ') {
+                    setShowMarkDown(false);
+                  }
+                }}
+              >
+                <MarkdownPreview source={descriptionField.value} />
+              </Box>
+            ) : (
+              <Box css={{ position: 'relative', width: '100%' }}>
+                <FormInputField
+                  id="description"
+                  name="description"
+                  control={control}
+                  css={{
+                    textarea: {
+                      resize: 'vertical',
+                      pb: '$xl',
+                      minHeight: 'calc($2xl * 2)',
+                    },
+                  }}
+                  areaProps={{
+                    autoFocus: true,
+                    onChange: e => {
+                      setValue('description', e.target.value);
+                    },
+                    onBlur: () => {
+                      if (
+                        descriptionField.value &&
+                        descriptionField.value.length > 0
+                      )
+                        setShowMarkDown(true);
+                    },
+                    onFocus: e => {
+                      e.currentTarget.setSelectionRange(
+                        e.currentTarget.value.length,
+                        e.currentTarget.value.length
+                      );
+                    },
+                  }}
+                  placeholder="What have you been working on?"
+                  textArea
+                />
+                <Text
+                  inline
+                  size="small"
+                  color="secondary"
+                  css={{
+                    position: 'absolute',
+                    right: '$sm',
+                    bottom: '$sm',
+                  }}
+                >
+                  Markdown Supported
+                </Text>
+              </Box>
+            )}
+            <Button
+              color="cta"
+              onClick={() => saveContribution(descriptionField.value)}
+            >
```

Comment:
If I double click the Add Contribution button I get an empty contribution so think we should add
```suggestion
              disabled={!descriptionField.value}
            >
```